zipline-bytecode/src/main/kotlin/app/cash/zipline/bytecode/JsObjectReader.kt

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline.bytecode
+
+import okio.Buffer
+import okio.BufferedSource
+import okio.Closeable
+import okio.IOException
+
+class JsObjectReader(
+  private val source: BufferedSource,
+) : Closeable by source {
+  constructor(byteArray: ByteArray) : this(Buffer().write(byteArray))
+
+  lateinit var atoms: AtomSet
+    private set
+
+  fun readJsObject(): JsObject {
+    check(!::atoms.isInitialized)
+    atoms = readAtoms()
+    return readObjectRecursive()
+  }
+
+  private fun readAtoms(): AtomSet {
+    val version = source.readByte().toInt()
+    if (version != version) {
```

Comment:
Should be `BC_VERSION`?
```suggestion
    if (version != BC_VERSION) {
```

Reply:
oooh yes definitely that. Fixed.